R/plot_var_compare.R

```diff
@@ -2,82 +2,118 @@
 #'
-#'run_glm(sim_folder)
+#'plot_var_compare(nc_file, field_file, 'temp', resample=FALSE) ##makes a plot
 #'
-#'plot_var_compare(nc_file, field_file, 'temp', resample=FALSE) ##makes a plot!
+#'#Change clor palette, custom legend title, save figure:
+#'plot_var_compare(nc_file, field_file, var_name = 'temp', resample = F, legend.title = 'Temp (*C)',
+#'color.palette = 'PuBuGn', color.direction = 1,fig_path = '~/Downloads/figtest.png', width = 6, height = 8, units = 'in')
 #'
 #'@importFrom akima interp
+#'@importFrom akima interp2xyz
+#'@importFrom gridExtra grid.arrange
+#'@author
+#'Jordan S. Read, Luke A. Winslow, Hilary A. Dugan
 #'@export
-plot_var_compare = function(nc_file, field_file, var_name, fig_path = FALSE, resample = TRUE, col_lim, ...){
+plot_var_compare = function(nc_file, field_file, var_name = 'temp', fig_path = NULL, resample = TRUE, 
+                            legend.title = NULL, interval = 1,method = 'match', text.size = 12,
+                            color.palette = 'RdYlBu', color.direction = -1, 
+                            obs.color = 'white', obs.alpha = 0.6, obs.shape = 16, obs.size = 1,...) {
   
   heatmaps <- .is_heatmap(nc_file, var_name)
   if (!heatmaps){
     warning('plot_var_compare not implemented for 1D variables')
     return()
   }
   
-  start_par = par(no.readonly = TRUE)
-  #Create layout
-  
-  mod_temp = get_var(nc_file, var_name, reference='surface')
-  mod_depths = get.offsets(mod_temp)
-  
+  surface <- get_surface_height(nc_file)
+  max_depth <- max(surface[, 2])
+  min_depth <- 0
+  z_out <- seq(min_depth, max_depth,by = interval) # Set plotting interval
+  mod_temp = get_var(nc_file, var_name, reference='surface',z_out = z_out)
+
+  data = resample_to_field(nc_file, field_file, var_name=var_name, method = method) 
   
-  data = resample_to_field(nc_file, field_file, var_name=var_name, ...)
-  if(resample){
-  	model_df <- resample_sim(mod_temp, t_out = unique(data$DateTime))
-  }else{
+  # Akima interpolation of observed data
+  dataClean = data %>% dplyr::filter_all(all_vars(!is.na(.)))
+
+  df_akima <-interp2xyz(interp(x=as.numeric(dataClean$DateTime), y=dataClean$Depth*1e6, z=dataClean$Observed_temp, duplicate="mean", linear = T,
+                               # xo=seq.POSIXt(min(dataClean$DateTime), max(dataClean$DateTime), by = 'day'),
+                               xo = as.numeric(seq(min(dataClean$DateTime), max(dataClean$DateTime), by = 'day')),
+                               yo = 1e6*seq(min(dataClean$Depth), max(dataClean$Depth), by = 1)), data.frame=TRUE) %>%
+    dplyr::mutate(x =  as.POSIXct(x, origin = '1970-01-01', tz = Sys.timezone())) %>%
+    dplyr::mutate(y = y/1e6) %>%
+    dplyr::arrange(x,y)
+
+  # Resample modeled data?
+  if(resample == TRUE) {
+  	model_df <-interp2xyz(interp(x=as.numeric(dataClean$DateTime), y=dataClean$Depth*1e6, z=dataClean$Modeled_temp, duplicate="mean", linear = T,
```

Comment:
Seems like a lot of redundant code w/ the interp above. Can/should this be broken out into a single function. One reason being, if you find a bug in this, you need to remember to change both.

Reply:
Good idea. 
I made a function in `timeseries_plots.R`

```
# Interpolate field or modeled data to grid 
# xcol, ycol, and zcol and column numbers from data.frame
# The spreads of ‘x’ and ‘y’ must be within four orders of magnitude of each other for ‘interp’ to work.
# Therefore must scale data to be within similar magnitude to numeric dates (1e6)
.interpolate2grid <- function(xyzData, xcol = 1, ycol = 2, zcol = 3) {
  gridData <-interp2xyz(interp(x = as.numeric(xyzData[,xcol]), y=xyzData[,ycol]*1e6, z=xyzData[,zcol], duplicate="mean", linear = T,
                               xo = as.numeric(seq(min(xyzData[,xcol]), max(xyzData[,xcol]), by = 'day')),
                               yo = 1e6*seq(min(xyzData[,ycol]), max(xyzData[,ycol]), by = 1)), data.frame=TRUE) %>%
    dplyr::mutate(x =  as.POSIXct(x, origin = '1970-01-01', tz = Sys.timezone())) %>%
    dplyr::mutate(y = y/1e6) %>%
    dplyr::arrange(x,y)
  
  return(gridData)
}
```